ui/components/multichain/detected-token-banner/detected-token-banner.js

```diff
@@ -0,0 +1,54 @@
+import React, { useContext } from 'react';
+import { useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { getDetectedTokensInCurrentNetwork } from '../../../selectors';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+import { EVENT, EVENT_NAMES } from '../../../../shared/constants/metametrics';
+import { BannerAlert } from '../../component-library';
+
+export const DetectedTokensBanner = ({
+  className,
+  setShowDetectedTokens,
+  ...props
+}) => {
+  const t = useI18nContext();
+  const trackEvent = useContext(MetaMetricsContext);
+
+  const detectedTokens = useSelector(getDetectedTokensInCurrentNetwork);
+  const detectedTokensDetails = detectedTokens.map(
+    ({ address, symbol }) => `${symbol} - ${address}`,
+  );
+
+  const handleOnClick = () => {
+    setShowDetectedTokens(true);
```

Comment:
I'm a little confused about passing down this callback as a prop? Shouldn't we be exposing the `actionButtonOnClick` prop like this? cc @darkwing 

```
const handleOnClick = () => {
  actionButtonOnClick && actionButtonOnClick();
```

Then during the use case of the component 
```
<DetectedTokensBanner actionButtonOnClick={() => setShowDetectedTokens(true)} />
```

It's like this prop can only accept this useState hook callback? Maybe we should at least mention that in the PropTypes? I primarily work with dumb components so lifting state is a bit rusty it just kind of smells to me...